src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ2735SSMConverter.cpp

```diff
@@ -0,0 +1,156 @@
+        }
+        return parseResult;
+    }
+
+    void JsonToJ2735SSMConverter::toJ2735SSM(const Json::Value &ssmDoc, std::shared_ptr<SignalStatusMessage> ssmPtr) const
+    {
+        try 
+        {        
+            ASN_STRUCT_FREE_CONTENTS_ONLY(asn_DEF_SignalStatusMessage, ssmPtr.get());
+            if (!ssmDoc.isMember("SignalStatus"))
+            {
+                return;
+            }
+
+            // populate SignalStatusMessage::second
+            if (ssmDoc["SignalStatus"].isMember("msOfMinute") && ssmDoc["SignalStatus"]["msOfMinute"].isNumeric())
+            {
+                ssmPtr->second = ssmDoc["SignalStatus"]["msOfMinute"].asInt64();
+            }
+
+            // populate SignalStatusMessage::timstamp
+            if (ssmDoc["SignalStatus"].isMember("minuteOfYear") && ssmDoc["SignalStatus"]["minuteOfYear"].isNumeric())
+            {
+                MinuteOfTheYear_t *timeStamp = (MinuteOfTheYear_t *)calloc(1, sizeof(MinuteOfTheYear_t));
+                *timeStamp = ssmDoc["SignalStatus"]["minuteOfYear"].asInt64();
+                ssmPtr->timeStamp = timeStamp;
+            }
+
+            SignalStatusList_t *statusPtr = (SignalStatusList_t *)calloc(1, sizeof(SignalStatusList_t));
+            SignalStatus *signalStatus = (SignalStatus *)calloc(1, sizeof(SignalStatus));
+
+            // populate SignalStatusMessage::status::id
+            if (ssmDoc["SignalStatus"].isMember("intersectionID") && ssmDoc["SignalStatus"]["intersectionID"].isNumeric())
+            {
+                signalStatus->id.id = ssmDoc["SignalStatus"]["intersectionID"].asInt64();
+            }
+
+            // populate SignalStatusMessage::status::sequenceNumber
+            if (ssmDoc["SignalStatus"].isMember("sequenceNumber") && ssmDoc["SignalStatus"]["sequenceNumber"].isNumeric())
+            {
+                signalStatus->sequenceNumber = ssmDoc["SignalStatus"]["sequenceNumber"].asInt64();
+            }
+
+            // populate SignalStatusMessage::status::sigStatus
+            if (ssmDoc["SignalStatus"].isMember("requestorInfo") && ssmDoc["SignalStatus"]["requestorInfo"].isArray())
+            {
+                Json::Value requesterJsonArr = ssmDoc["SignalStatus"]["requestorInfo"];
+                for (auto itr = requesterJsonArr.begin(); itr != requesterJsonArr.end(); itr++)
+                {
+                    SignalStatusPackage *signalStatusPackage = (SignalStatusPackage *)calloc(1, sizeof(SignalStatusPackage));
+                    populateSigStatusPackage(signalStatusPackage, itr);
+                    asn_sequence_add(&signalStatus->sigStatus.list.array, signalStatusPackage);
+                } // Populate signal status package
+            }
+
+            asn_sequence_add(&statusPtr->list.array, signalStatus);
+            ssmPtr->status = *statusPtr;
+        }
+        catch(exception &ex)
+        {
+            std::cout << "ERROR: Cannot read JSON file." << std::endl;
```

Comment:
Please correct this to use PLOG macros and this is not a JSON file it is a JSON message.

Reply:
https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/533/commits/53ec358d4de07c0a7567a9b5dfe7e86e0a5624c3